### ISSUE TITLE
bug in ddpm scheduler

### ISSUE THREAD

**Issue by LSQsjtu** [huggingface/diffusers#3089]
### Describe the bug

forget to convert self.alphas_cumprod to the same device with samples
https://github.com/huggingface/diffusers/blob/46c52f9b9607e6ecb29c782c052aea313e6487b7/src/diffusers/schedulers/scheduling_ddpm.py#L326, which will cause calculation error.
 
suggest add line self.alphas_cumprod = self.alphas_cumprod.to(device=sample.device, dtype=sample.dtype).

### Reproduction

use DDPMScheduler.step()

### System Info

0.15.0.dev0

**Comment by patrickvonplaten** [comment]
Hey @LSQsjtu, 

could you make a reproducible code snippet that we could copy-paste into a Python interpreter? :-)

**Comment by LSQsjtu** [comment]
```python
import torch
from diffusers import DDPMScheduler

noise_scheduler = DDPMScheduler(
    beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", num_train_timesteps=1000
)

device = torch.device('cuda')
noisy_latent = torch.ones((1, 3, 64, 64)).to(device)
noise = torch.ones((1, 3, 64, 64)).to(device)

noise_scheduler.step(noise, 500, noisy_latent)
```

I am sorry, it seems strange that in this way I run correctly. But the variable beta_prod_t is cpu variable and the model_output is cuda variable. https://github.com/huggingface/diffusers/blob/3bf5ce21ad2fd39c0443f8f689e12761c0f67a0f/src/diffusers/schedulers/scheduling_ddpm.py#L336
And it didn't throw any error.

**Comment by patrickvonplaten** [comment]
> ```python
> import torch
> from diffusers import DDPMScheduler
> 
> noise_scheduler = DDPMScheduler(
>     beta_start=0.00085, beta_end=0.012, beta_schedule="scaled_linear", num_train_timesteps=1000
> )
> 
> device = torch.device('cuda')
> noisy_latent = torch.ones((1, 3, 64, 64)).to(device)
> noise = torch.ones((1, 3, 64, 64)).to(device)
> 
> noise_scheduler.step(noise, 500, noisy_latent)
> ```

I think this behavior is expected. Both inputs are CUDA so I don't think that this should throw an error

**Comment by LSQsjtu** [comment]
Thanks
